other-examples/confluentcloud/collector.yaml

```diff
@@ -0,0 +1,57 @@
+receivers:
+  kafkametrics:
+    brokers:
+      - $CLUSTER_BOOTSTRAP_SERVER
+    protocol_version: 2.0.0
+    scrapers:
+      - brokers
+      - topics
+      - consumers
+    auth:
+      sasl:
+        username: $CLUSTER_API_KEY
+        password: $CLUSTER_API_SECRET
+        mechanism: PLAIN
```

Comment:
Could this expose plaintext credentials connecting to Confluent? Is there a more secure mechanism you could use?

Reply:
Yes, they could use TLS certs, which would be the better way for a prod setup. This was a bit more simplistic for the purposes of an example setup, but I could change it if you think that makes more sense.

Reply:
My only concern is that people may not think deeply about what they're doing when picking up this example and running it. In the worst case someone might test it using credentials that really shouldn't be exposed.

Putting otel and the collector aside, not being very familiar with conflent cloud, I assume they have similar getting-started tutorials. How, if at all, do their tutorials protect against leaking credentials?

Reply:
hmm, while it is plain text the suggestion is to set it as an environment variable, not hard set the value as part of the collector. My impression was that having the secret stored in a local environment variable should be safe for an example collector setup, while in production they might set that to something like a vault secret location.

Maybe I could add that for production setups they should read through confluent's docs on secret management?
https://docs.confluent.io/platform/current/security/secrets.html

Reply:
The otel docs suggest using key.pem files. I think that makes sense as well, I can update this to use the suggested otel configuration:

`    auth:
      tls:
        ca_file: ca.pem
        cert_file: cert.pem
        key_file: key.pem
    collection_interval: 5s`